Infrastructure.STS.OrganizationSystem/DomainServices/StsOrganizationSystemService.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.ServiceModel;
+using Core.Abstractions.Types;
+using Core.DomainModel.Organization;
+using Core.DomainServices.Model.StsOrganization;
+using Core.DomainServices.Organizations;
+using Core.DomainServices.SSO;
+using Infrastructure.STS.Common.Factories;
+using Infrastructure.STS.Common.Model;
+using Infrastructure.STS.Common.Model.Client;
+using Infrastructure.STS.OrganizationSystem.OrganisationSystem;
+using Serilog;
+
+namespace Infrastructure.STS.OrganizationSystem.DomainServices
+{
+    public class StsOrganizationSystemService : IStsOrganizationSystemService
+    {
+        private readonly IStsOrganizationService _organizationService;
+        private readonly ILogger _logger;
+        private readonly string _certificateThumbprint;
+        private readonly string _serviceRoot;
+
+        public StsOrganizationSystemService(IStsOrganizationService organizationService, StsOrganisationIntegrationConfiguration configuration, ILogger logger)
+        {
+            _organizationService = organizationService;
+            _logger = logger;
+            _certificateThumbprint = configuration.CertificateThumbprint;
+            _serviceRoot = $"https://{configuration.EndpointHost}/service/Organisation/OrganisationSystem/5";
+        }
+
+        public Result<ExternalOrganizationUnit, DetailedOperationError<ResolveOrganizationTreeError>> ResolveOrganizationTree(Organization organization)
+        {
+            //Search for org units by org uuid
+            using var clientCertificate = X509CertificateClientCertificateFactory.GetClientCertificate(_certificateThumbprint);
```

Comment:
**This is a big beast** but to not introduce too much risk (I know much of this is carried over from the old service) we should not retain it as is. It surprises me just how compatible the code was